### PR TITLE
Visual update: the rendering of suggestions

### PR DESCRIPTION
**Pull request by isoos**
- deployed to [staging](https://dartlang-pub-dev.appspot.com/)

- visual upgrades include icon-like things and a subtle hover effect, e.g. a [bad package](https://dartlang-pub-dev.appspot.com/packages/db_executor#-analysis-tab-)

- enables additional help page (#1354), e.g. [http](https://dartlang-pub-dev.appspot.com/packages/db_executor#-analysis-tab-) packages' last entry has a link that is now customizable via the pub site's codebase

### CODE REVIEW

File: app/test/frontend/golden/analysis_tab_http.html [hunk 1]
```diff
@@ -78,49 +78,83 @@ <h4>Platforms</h4>
...
-      <blockquote><p>Make sure <code>dartdoc</code> runs without any issues.</p>
-</blockquote>
-    </li>
-    <li>
-      <div><p>Fix analysis and formatting issues.</p>
+<div class="suggestion-row">
+  <div class="suggestion-title">
+    <span class="suggestion-icon-danger"></span>
+    <p>Running <code>dartdoc</code> failed.</p>
+
```
Comment (sortie) [inline app/test/frontend/golden/analysis_tab_http.html]: Can we figure out why these needless blank lines are being inserted?

File: app/test/frontend/golden/analysis_tab_http.html [hunk 2]
```diff
@@ -78,49 +78,83 @@ <h4>Platforms</h4>
...
-    <li>
-      <div><p>Maintain an example.</p>
+<div class="suggestion-row">
+  <div class="suggestion-title">
+    <span class="suggestion-icon-info"></span>
+    <p>Package is pre-v1 release.</p>
+
+  </div>
+  <div class="suggestion-description">
+    <p>While there is nothing inherently wrong with versions of <code>0.*.*</code>, it usually means that the author is still experimenting with the general direction API.</p>
```
Comment (sortie) [inline app/test/frontend/golden/analysis_tab_http.html]: s/general direction API/general direction of the API/g?

**Comment by sortie** [comment]
Nevermind; you said you put it into staging. Looks good. Not sure if the red box should also be a circle.

**Comment by isoos** [comment]
> I didn't review how this renders to html. Is there a screenshot?

Follow the link to staging, and search for random packages. I can update screenshot here tool.

File: app/test/frontend/golden/analysis_tab_http.html (hunk 2 above)
Comment (isoos) [reply app/test/frontend/golden/analysis_tab_http.html]: Coming from pana, created an issue for it: https://github.com/dart-lang/pana/issues/360

File: app/test/frontend/golden/analysis_tab_http.html (hunk 1 above)
Comment (isoos) [reply app/test/frontend/golden/analysis_tab_http.html]: It is a combination of:
- the output of `markdown`,
- the output of `mustache`,
- keeping our templates sane.

Unfortunately we have a lot of these, I'll revisit that later (https://github.com/dart-lang/pub-dartlang-dart/issues/1418)

**Comment by isoos** [comment]
> Not sure if the red box should also be a circle.

I wanted that to be stand out, apart from the others. We can tweak it later on I guess. I have another coming PR related to the analysis tab, I'd suggest to tweak it together with that.

**Comment by mit-mit** [comment]
I like the indicators a lot! But why is something like failing analysis just a blue info circle?
https://dartlang-pub-dev.appspot.com/packages/bytes_data_builder#-analysis-tab-

**Comment by mit-mit** [comment]
>> Not sure if the red box should also be a circle.

> I wanted that to be stand out, apart from the others. We can tweak it later on I guess. I have another coming PR related to the analysis tab, I'd suggest to tweak it together with that.

Are warnings not usually triangle shaped?

**Comment by isoos** [comment]
> But why is something like failing analysis just a blue info circle?
> https://dartlang-pub-dev.appspot.com/packages/bytes_data_builder#-analysis-tab-

That analysis has no failing component in it. The code quality is 80+/100, and the suggestions are mostly related to non-critical items. Does any of it stand out as something we should mark as a warning or error?

**Comment by mit-mit** [comment]
> That analysis has no failing component in it. The code quality is 80+/100, and the suggestions are mostly related to non-critical items. Does any of it stand out as something we should mark as a warning or error?

Ah, I understand. I think we should change the heading for those to 'Fix analysis and formatting hints.'

**Comment by isoos** [comment]
> I think we should change the heading for those to 'Fix analysis and formatting hints.'

That is already changing in https://github.com/dart-lang/pana/pull/358 and it becomes: "Additional issues in the following files (4 errors, 8 hints) ..."

**Comment by mit-mit** [comment]
I suggest 'Fix analysis and formatting errors|hints|errors and hints'

**Comment by isoos** [comment]
> Are warnings not usually triangle shaped?

I've updated the warning sign to a triangle:

screen shot 2018-06-28 at 15 44 51

With that, the error's red square resembles a "stop sign", the warning is a warning triangle, and hints are only info buttons. I'll have another visual update for that page soon, I'll merge this, and let's iterate on both of these, not separately.

**Pull request merged**